src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -435,5 +435,13 @@ public bool IsTypeDefinition
                 return GetTypeDefinition() == this;
             }
         }
+
+        /// <summary>
+        /// Determine if two types share the same type definition
+        /// </summary>
+        public bool HasSameTypeDefinition(DefType otherType)
```

Comment:
Shouldn't `otherType` be a `TypeDesc`?

And if not, would this method make more sense on `DefType`?